internal/trie/node/copy.go

```diff
@@ -3,76 +3,107 @@
 
 package node
 
+// CopySettings contains settings to configure the deep copy
+// of a node. By default, it:
+// - does not deep copy children recrursively
+// - does not copy cached fields HashDigest and Encoding
+// - deep copies the key field
+// - deep copies the value field
+type CopySettings struct {
```

Comment:
it would be nice to also have a `DefaultCopySettings` variable potentially?

Reply:
Added `DefaultCopySettings() CopySettings` for this, and also changed field names

- `LeaveKeyEmpty` ==> `CopyKey` (inverted)
- `LeaveValueEmpty` ==> `CopyValue` (inverted)

so now the empty CopySettings struct does not copy anything, the dev has to use `node.DefaultCopySettings()` or fill what needs to be copied.